### PR TITLE
add royalty calculation method and RPC

### PR DESCRIPTION
**Pull request by Quexington**
https://github.com/Chia-Network/chia-blockchain/issues/12836

### CODE REVIEW

**Comment by Quexington** [comment]
Example request:
```
{
  "royalty_assets": [
    {
      "asset": "0xabc...",
      "royalty_address": "0x123...",
      "royalty_pts": 5000,
    },
  ],
  "fungible_assets": [
    {
      "asset": 1,  # can be wallet ID or hex or whatever, it's not going to be used or edited
      "amount": 200,
    }
  ],
}
```
Example response:
```
{
  "0xabc...": [
    {
      "asset": 1,
      "address": "0x123...",
      "amount": 100,
    },
  ],
}
```

**Comment by trepca** [comment]
Do CLI trading commands need to be updated to use this too or is that already using the underlying backend?

**Comment by Quexington** [comment]
> Do CLI trading/offer commands need to be updated to use this too or is that already using the underlying backend?

I don't think there's really any CLI around royalties right now just a warning.  We could certainly improve on that.

File: chia/rpc/wallet_rpc_client.py [hunk 1]
```diff
@@ -677,6 +677,21 @@ async def add_uri_to_nft(self, wallet_id, nft_coin_id, key, uri, fee):
...
         return response
 
+    async def nft_calculate_royalties(
+        self,
+        royalty_assets_dict: Dict[Any, Tuple[Any, uint16]],
+        fungible_asset_dict: Dict[Any, uint64],
+    ) -> Dict[Any, List[Dict[str, Any]]]:
+        request: Dict[str, Any] = {
+            "royalty_assets": [
+                {"asset": id, "royalty_address": royalty_info[0], "royalty_pts": royalty_info[1]}
```
Comment (paninaro) [inline chia/rpc/wallet_rpc_client.py:687]: Can we rename `royalty_pts` to `royalty_percentage` to be consistent with the existing field in nft_info?

**Review by paninaro — CHANGES_REQUESTED** [review]
Looks good. Just the one request to rename `royalty_pts`

**Comment by Quexington** [comment]
@paninaro I fixed your comment, but also added another commit for fixing the CLI related to what @trepca said as well

**Review by paninaro — DISMISSED** [review]
lgtm

**Review by trepca — APPROVED** [review]

**Pull request merged**